libvast/builtins/formats/zeek_tsv.cpp

```diff
@@ -0,0 +1,757 @@
+    auto r = caf::get<record_type>(t);
+    for (const auto& [_, offset] : r.leaves())
+      header.append(fmt::format("{}{}", sep, to_string(r.key(offset))));
+    header.append("\n#types");
+    for (const auto& [field, _] : r.leaves())
+      header.append(fmt::format("{}{}", sep, to_zeek_string(field.type)));
+    out = std::copy(header.begin(), header.end(), out);
+    return true;
+  }
+
+  template <typename It>
+  auto print_values(It& out, const view<record>& x) const noexcept -> bool {
+    auto first = true;
+    for (const auto& [_, v] : x) {
+      if (not first) {
+        ++out = sep;
+      } else {
+        first = false;
+      }
+      caf::visit(visitor{out, *this}, v);
+    }
+    return true;
+  }
+
+  template <typename It>
+  auto print_closing_line(It& out) const noexcept -> void {
+    if (not disable_timestamp_tags) {
+      out = fmt::format_to(out, "#close{}{}\n", sep, generate_timestamp());
+    }
+  }
+
+  template <class Iterator>
+  struct visitor {
+    visitor(Iterator& out, const zeek_printer& printer)
+      : out{out}, printer{printer} {
+    }
+
+    auto operator()(caf::none_t) noexcept -> bool {
+      out = std::copy(printer.unset_field.begin(), printer.unset_field.end(),
+                      out);
+      return true;
+    }
+
+    auto operator()(auto x) noexcept -> bool {
+      make_printer<decltype(x)> p;
+      return p.print(out, x);
+    }
+
+    auto operator()(view<bool> x) noexcept -> bool {
+      return printers::any.print(out, x ? 'T' : 'F');
+    }
+
+    auto operator()(view<pattern>) noexcept -> bool {
+      die("unreachable");
+    }
+
+    auto operator()(view<map>) noexcept -> bool {
+      die("unreachable");
+    }
+
+    auto operator()(view<std::string> x) noexcept -> bool {
+      if (x.empty()) {
+        out = std::copy(printer.unset_field.begin(), printer.unset_field.end(),
+                        out);
+        return true;
+      }
+      for (auto c : x)
+        if (std::iscntrl(c) or c == printer.sep or c == printer.set_sep) {
+          auto hex = detail::byte_to_hex(c);
+          *out++ = '\\';
+          *out++ = 'x';
+          *out++ = hex.first;
+          *out++ = hex.second;
+        } else {
+          *out++ = c;
+        }
+      return true;
+    }
+
+    auto operator()(const view<list>& x) noexcept -> bool {
+      if (x.empty()) {
+        out = std::copy(printer.empty_field.begin(), printer.empty_field.end(),
+                        out);
+        return true;
+      }
+      auto first = true;
+      for (const auto& v : x) {
+        if (not first) {
+          ++out = printer.set_sep;
+        } else {
+          first = false;
+        }
+        caf::visit(*this, v);
+      }
+      return true;
+    }
+
+    auto operator()(const view<record>& x) noexcept -> bool {
+      // TODO: This won't be needed when flatten() for table_slices is in the
+      // codebase.
+      VAST_WARN("Printing records as Zeek data is currently a work in "
```

Comment:
```suggestion
      VAST_WARN("printing records as Zeek data is currently a work in "
```

Reply:
Fixed in https://github.com/tenzir/vast/pull/3114/commits/d01e63a216f6cf0cb80a82adb3a4ae4e89a9286c